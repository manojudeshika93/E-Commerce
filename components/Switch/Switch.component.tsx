import React from 'react';
import { Switch as RNSwitch, SwitchProps } from 'react-native-switch';

import { Color, tw } from '@/config/tw';
import { Text, View } from 'react-native';

export function Switch({ disabled = false, value, onValueChange }: Readonly<SwitchProps>) {
  const toggleWidth = 52;
  const toggleHeight = 27;
  const circleSize = 22;
  const switchWidthMultiplier = toggleWidth / circleSize;

  return (
    <View style={tw`flex-row items-center justify-center gap-2 bg-custom-white p-2 rounded-full`}>
      <Text style={tw`text-b1-semibold text-secondary-600`}>EN</Text>
      <RNSwitch
        testID='switch'
        value={value}
        containerStyle={tw``}
        outerCircleStyle={tw`mx-0.5`}
        onValueChange={onValueChange}
        disabled={disabled}
        circleSize={circleSize}
        barHeight={toggleHeight}
        circleBorderWidth={0}
        backgroundActive={disabled ? Color.secondary[300] : Color.secondary[400]}
        backgroundInactive={disabled ? Color.secondary[300] : Color.secondary[400]}
        circleActiveColor={disabled ? Color.secondary[400] : Color.primary[500]}
        circleInActiveColor={disabled ? Color.secondary[400] : Color.primary[500]}
        renderActiveText={false}
        renderInActiveText={false}
        switchLeftPx={switchWidthMultiplier}
        switchRightPx={1.6}
        switchWidthMultiplier={switchWidthMultiplier}
      />
      <Text style={tw`text-b1-semibold text-secondary-600`}>AR</Text>
    </View>
  );
}
