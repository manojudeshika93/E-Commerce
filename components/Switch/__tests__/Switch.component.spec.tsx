import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';

import { Switch } from '../Switch.component';

describe('Switch Component', () => {
  const onValueChangeMock = jest.fn();

  const getRenderedComponent = (props = {}) =>
    render(<Switch value={false} onValueChange={onValueChangeMock} {...props} />);

  it('should render the Switch component correctly', () => {
    const { toJSON } = getRenderedComponent();
    expect(toJSON()).toMatchSnapshot();
  });

  it('should display EN and AR text labels', () => {
    const { getByText } = getRenderedComponent();
    expect(getByText('EN')).toBeTruthy();
    expect(getByText('AR')).toBeTruthy();
  });

  it('should call onValueChange when toggled', () => {
    const { getByTestId } = getRenderedComponent();
    const switchElement = getByTestId('switch');

    fireEvent(switchElement, 'valueChange', true);
    expect(onValueChangeMock).toHaveBeenCalledWith(true);

    fireEvent(switchElement, 'valueChange', false);
    expect(onValueChangeMock).toHaveBeenCalledWith(false);
  });
});
