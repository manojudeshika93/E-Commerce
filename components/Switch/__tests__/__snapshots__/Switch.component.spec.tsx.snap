// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch Component should render the Switch component correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 9999,
      "flexDirection": "row",
      "gap": 8,
      "justifyContent": "center",
      "paddingBottom": 8,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 8,
    }
  }
>
  <Text
    style={
      {
        "color": "#404040",
        "fontFamily": "Inter_600SemiBold",
        "fontSize": 16,
        "lineHeight": 24,
      }
    }
  >
    EN
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "rgba(160, 160, 160, 1)",
        "borderRadius": 22,
        "height": 27,
        "width": 52,
      }
    }
    testID="switch"
  >
    <View
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "left": -13.75,
          "marginLeft": 2,
          "marginRight": 2,
          "width": 52,
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "rgba(195, 1, 69, 1)",
            "borderColor": "rgba(80, 80, 80, 1)",
            "borderRadius": 11,
            "borderWidth": 0,
            "height": 22,
            "justifyContent": "center",
            "width": 22,
          }
        }
      />
    </View>
  </View>
  <Text
    style={
      {
        "color": "#404040",
        "fontFamily": "Inter_600SemiBold",
        "fontSize": 16,
        "lineHeight": 24,
      }
    }
  >
    AR
  </Text>
</View>
`;
